evacuees/src/UI/embc-evacuee/src/app/sharedModules/components/dashboard/dashboard.component.ts

```diff
@@ -91,39 +92,46 @@ export class DashboardComponent implements OnInit {
   showActiveList = true;
   showInactiveList = true;
   currentChild: EvacuationCardComponent;
+  tabIndex: number;
 
   evacuatedFrom?: string;
-  referenceNumber: string;
 
 
   constructor(
     private route: ActivatedRoute, private dataService: DataService, public formCreationService: FormCreationService,
-    private router: Router, private dialogService: DialogService) { }
+    private router: Router, private dialogService: DialogService, private cacheService: CacheService) { }
 
 
   ngOnInit(): void {
     this.currentFlow = this.route.snapshot.data.flow;
     this.evacuatedFrom = this.dataSourceActive[this.dataSourceActive.length - 1]?.from;
 
     this.router.events.pipe(
-      filter((event: RouterEvent) => event instanceof NavigationEnd),
-      debounceTime(500)
-    ).subscribe(() =>
-      this.openReferenceNumberPopup()
-    );
-
+      filter((event: RouterEvent) => event instanceof ActivationEnd),
+      first()
+    ).subscribe((event: any) => {
+      this.cacheService.set('previousComponent', event.snapshot.component.name);
+    });
+
+    if (this.cacheService.get('previousComponent') === 'EditComponent') {
+      this.tabIndex = 2;
+    } else if (this.cacheService.get('previousComponent') === 'ConfirmRestrictionComponent') {
```

Comment:
Do you need else if condition if you are doing null check against reference number?